neptune_tensorflow_keras/impl/__init__.py

```diff
@@ -112,22 +112,28 @@ def __init__(self, run: Run, base_namespace: Optional[str] = None):
 
         run[INTEGRATION_VERSION_KEY] = __version__
 
-    def _log_metrics(self, logs, trigger):
+    def _log_metrics(self, logs, category: str, trigger: str):
         if not logs:
             return
 
-        logger = self._metric_logger[trigger]
+        logger = self._metric_logger[category][trigger]
 
         for metric, value in logs.items():
             try:
-                if metric in ('batch', 'size'):
+                if metric in ('batch', 'size') or metric.startswith('val_'):
                     continue
                 logger[metric].log(value)
             except NeptuneException:
                 pass
 
-    def on_batch_end(self, batch, logs=None):  # pylint:disable=unused-argument
-        self._log_metrics(logs, 'batch')
+    def on_train_batch_end(self, batch, logs=None):  # pylint:disable=unused-argument
```

Comment:
To była publiczna metoda. Nie psujemy kompatybilności wstecznej w ten sposób?

Reply:
Ta metoda jest uruchamiana w szczególności przez tą poprzednią. To poliformizm a nie nasze API jak coś :wink: